jerry-core/parser/js/js-scanner-util.c

```diff
@@ -1738,6 +1731,96 @@ scanner_create_unused_literal (parser_context_t *context_p, /**< context */
   context_p->literal_count++;
 } /* scanner_create_unused_literal */
 
+#if ENABLED (JERRY_ES2015)
+/**
+ * Emit checks for redeclared bindings in the global lexical scope.
+ */
+void
+scanner_check_variables (parser_context_t *context_p) /**< context */
+{
+  scanner_info_t *info_p = context_p->next_scanner_info_p;
+  const uint8_t *next_data_p = (const uint8_t *) (info_p + 1);
+  lexer_lit_location_t literal;
+
+  JERRY_ASSERT (info_p->type == SCANNER_TYPE_FUNCTION);
+
+  literal.char_p = info_p->source_p - 1;
+
+  while (next_data_p[0] != SCANNER_STREAM_TYPE_END)
+  {
+    uint32_t type = next_data_p[0] & SCANNER_STREAM_TYPE_MASK;
+    const uint8_t *data_p = next_data_p;
+
+    if (type == SCANNER_STREAM_TYPE_HOLE)
```

Comment:
I think this type is not allowed in global scope. I would add an assert instead.